src/commands/build.cr

```diff
@@ -44,9 +44,12 @@ module Shards
         Log.debug { "#{Shards.crystal_bin} #{args.join(' ')}" }
 
         error = IO::Memory.new
-
         status = Process.run(Shards.crystal_bin, args: args, output: Process::Redirect::Inherit, error: error)
-        raise Error.new("Error target #{target.name} failed to compile:\n#{error}") unless status.success?
+        unless status.success?
+          raise Error.new("Error target #{target.name} failed to compile:\n#{error}")
+        else
+          STDERR << error if error.size > 0
```

Comment:
I'd suggest to use `puts` for the odd chance that `error` misses a trailing newline.
Size comparison can be replaced by `empty?`.
```suggestion
          STDERR.puts error unless error.empty?
```